### PR TITLE
Airship

### PR DESCRIPTION
**Pull request by js6pak**

### CODE REVIEW

**Comment by IrishBruse** [comment]
Aren't publics on interface methods redundant?

**Review by miniduikboot — CHANGES_REQUESTED** [review]
This branch has gotten a bit too large, making it hard and not fun to review.

Most of my critique concerns the vent api, which is something that imo should not have been in this PR as it is not really airship related

Removing the reactor stuff was also something that was bound to happen in future, but the timing is rather unfortunate as the replacement is not ready yet.

I think that in future we should aim to keep these update branches smaller and do non-essential refactorings afterwards.

File: src/Impostor.Api/Events/Game/Player/IPlayerVentEvent.cs [hunk 1]
```diff
@@ -7,11 +7,6 @@ public interface IPlayerVentEvent : IPlayerEvent
         /// <summary>
         ///     Gets get the id of the used vent.
         /// </summary>
-        public VentLocation VentId { get; }
-
-        /// <summary>
-        ///     Gets a value indicating whether the vent was entered or exited.
-        /// </summary>
-        public bool VentEnter { get; }
+        public Vent Vent { get; }
```
[the remark this answers is not in the document]
Comment (js6pak) [reply src/Impostor.Api/Events/Game/Player/IPlayerVentEvent.cs:10]: I should've added comment to that event, but:
- IPlayerVentEvent: called when player vents (SnapTos) to another vent
- IPlayerEnterVentEvent/IPlayerExitVentEvent: called when player enters/exits a vent

File: src/Impostor.Api/Innersloth/Customization/ColorType.cs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ﻿namespace Impostor.Api.Innersloth.Customization
 {
-    public enum ColorType : byte
+    public enum ColorType
```
[the remark this answers is not in the document]
Comment (js6pak) [reply src/Impostor.Api/Innersloth/Customization/ColorType.cs:3]: To reflect airship client changes, it was changed from byte to int there aswell

File: src/Impostor.Api/Innersloth/Maps/AirshipData.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            ConnectVents(AirshipVent.Ids.ViewingDeck, left: AirshipVent.Ids.Cockpit);
+            ConnectVents(AirshipVent.Ids.EngineRoom, left: AirshipVent.Ids.Kitchen, right: AirshipVent.Ids.MainHallBottom);
+            ConnectVents(AirshipVent.Ids.Kitchen, left: AirshipVent.Ids.EngineRoom, right: AirshipVent.Ids.MainHallBottom);
+            ConnectVents(AirshipVent.Ids.MainHallBottom, left: AirshipVent.Ids.EngineRoom, right: AirshipVent.Ids.Kitchen);
+            ConnectVents(AirshipVent.Ids.GapRight, left: AirshipVent.Ids.MainHallTop, right: AirshipVent.Ids.GapLeft);
+            ConnectVents(AirshipVent.Ids.GapLeft, left: AirshipVent.Ids.MainHallTop, right: AirshipVent.Ids.GapRight);
+            ConnectVents(AirshipVent.Ids.MainHallTop, left: AirshipVent.Ids.GapLeft, right: AirshipVent.Ids.GapRight);
+            ConnectVents(AirshipVent.Ids.Showers, left: AirshipVent.Ids.Records, right: AirshipVent.Ids.CargoBay);
+            ConnectVents(AirshipVent.Ids.Records, left: AirshipVent.Ids.Showers, right: AirshipVent.Ids.CargoBay);
+            ConnectVents(AirshipVent.Ids.CargoBay, left: AirshipVent.Ids.Showers, right: AirshipVent.Ids.Records);
```
[the remark this answers is not in the document]
Comment (js6pak) [reply src/Impostor.Api/Innersloth/Maps/AirshipData.cs:39]: I can't really think of any better way, but it definitely shouldn't be editing ReadOnlyDictionary, you are right on that

File: src/Impostor.Api/Innersloth/Platforms.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+namespace Impostor.Api.Innersloth
+{
+    public enum Platforms
```
[the remark this answers is not in the document]
Comment (js6pak) [reply src/Impostor.Api/Innersloth/Platforms.cs:3]: wdym?

File: src/Impostor.Server/Net/Inner/Objects/Components/InnerPlayerPhysics.cs [hunk 5]
```diff
@@ -38,32 +38,56 @@ public override ValueTask DeserializeAsync(IClientPlayer sender, IClientPlayer?
...
+
+                    var vent = Game.GameNet.ShipStatus!.Data.Vents[ventId];
+
+                    switch (call)
+                    {
+                        case RpcCalls.EnterVent:
+                            await _eventManager.CallAsync(new PlayerEnterVentEvent(Game, sender, _playerControl, vent));
+                            break;
+                        case RpcCalls.ExitVent:
+                            await _eventManager.CallAsync(new PlayerExitVentEvent(Game, sender, _playerControl, vent));
```
[the remark this answers is not in the document]
Comment (js6pak) [reply src/Impostor.Server/Net/Inner/Objects/Components/InnerPlayerPhysics.cs:77]: https://github.com/Impostor/Impostor/pull/377#discussion_r610910629
https://github.com/Impostor/Impostor/pull/377/files#diff-29d84eef1afe966f93dec1a9ebc0f21ce87727e8152397dbdafd8f2f58f1db28R131

File: src/Impostor.Api/Events/Game/Player/IPlayerVentEvent.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (miniduikboot) [reply src/Impostor.Api/Events/Game/Player/IPlayerVentEvent.cs:10]: Ah, didn't get that distinction. Maybe it'd me more clear to rename IPlayerVentEvent to IPlayerMoveEvent?

Adding comments to the would be much appreciated, but that's maybe out of scope for this PR

File: src/Impostor.Api/Innersloth/Maps/AirshipData.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (miniduikboot) [reply src/Impostor.Api/Innersloth/Maps/AirshipData.cs:39]: Would it be possible to add the left/right/center optional arguments to the class constructors?

File: src/Impostor.Api/Innersloth/Platforms.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (miniduikboot) [reply src/Impostor.Api/Innersloth/Platforms.cs:3]:
```
Unknown = 0
StandaloneEpicPc = 1,
// etc
```

File: src/Impostor.Server/Net/Inner/Objects/Components/InnerPlayerPhysics.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (miniduikboot) [reply src/Impostor.Server/Net/Inner/Objects/Components/InnerPlayerPhysics.cs:77]: disregard this, I missed the different purpose of playerventevent

**Pull request merged**